arduino/smartcar/smartcar.ino

```diff
@@ -1,9 +1,45 @@
-#include <Smartcar.h>
+#include <SmartCar.h>
+
+const int triggerPin = 6;
+const int echoPin = 7;
+const unsigned int maxDistance = 100;
+
+ArduinoRuntime arduinoRuntime;
+
+BrushedMotor leftMotor(arduinoRuntime, smartcarlib::pins::v2::leftMotorPins);
+BrushedMotor rightMotor(arduinoRuntime, smartcarlib::pins::v2::rightMotorPins);
+
+DifferentialControl control(leftMotor, rightMotor);
+
+SR04 frontSensor(arduinoRuntime, triggerPin, echoPin, maxDistance);
+
+SimpleCar car(control);
+
+const auto forwardSpeed = 30;
+const auto stopSpeed = 0;
+
+void stopCar(){
+    car.setSpeed(stopSpeed);
+}
+void startCar(){
+    car.setSpeed(forwardSpeed);
+}
 
 void setup() {
-  // put your setup code here, to run once:
+
+    Serial.begin(9600);
 }
 
 void loop() {
-  // put your main code here, to run repeatedly:
+
+    Serial.println(frontSensor.getDistance());
+    delay(100);
+    if(frontSensor.getDistance() >= 50){
```

Comment:
OK, but what happens when the sensor doesn't detect any obstacle? It will also stop.

Reply:
That is strange, because on my smce the car is driving forward and stops only when it gets values above 50 and that are equal to 50. So the car still drives when the sensor gives outputs of 0.

Reply:
Ah right, the way you have written this is very confusing.

OK, what happens then if your sensor detects 40? Shouldn't it stop?

Reply:
I'll let you think of the solution, it's not that hard 😄

Reply:
So then it is the other way around, if the distance is less than or equal to 50, and not equal to 0, then the car should stop

Reply:
👍